WIP create page refactor

krayzieeddi opened the pull request:
NSC-events-nextjs_sprint22_<issue#>_Implementing_new_create_page_styling 


## Summary & Changes 📃
- Started implementing the new Figma mock-up design on the NSC event's create page
- Have the styling down, but need to tackle the functionality like proper dark mode and required fields after submitting
- Got rid of event categories as its was deemed redundant

- **Summary:** (Briefly describe what this PR does)
  - This PR is to refactor the styling of the create page and its dark mode version while changing the mandatory required fields that the user must enter for the form to be able to submit.
  - 
- **Changes:**
  - Started refactoring the containers and aligned the text fields

## Screenshots / Visual Aids 🔎



## How to Test 🧪
1. To see changes, just start up this branches' back and front end, then navigate to the create event page


## Checklist ✅

- [ ] Refactored the styling of the page
- [ ] Making sure that the dark mode works properly with the new styling
- [ ] Changing required fields to optional

Asfand00 reviewed (commented):
Hey @krayzieeddi, Great work so far on the design, I have a lot of feedback on these. 

Firstly, the design on the light mode is good, some minor changes are needed:
- There's an asterisk on the Title text field that needs to be removed (see image)
- Both eventCoverPhoto and eventDocument text fields need to be removed as they are no longer needed on the event form (I'm currently working on this issue here: https://github.com/SeattleColleges/nsc-events-nextjs/issues/652)
- The tags section color doesn't seem to fit or color coordinate with the other elements. Might be best to keep the background color a baby blue or light blue and then change the tag buttons to white similar to the Figma design.  

figma design page:
![image](https://github.com/user-attachments/assets/fb426273-aa25-4b60-848e-e8293da3c708)

the implemented design page (with my edits):
![image](https://github.com/user-attachments/assets/e34b469f-e881-4a72-abfd-25d72ab82bbb)

For the dark mode, there are some issues too:
-  The way you are applying styling for the dark mode is affecting all elements on every page not just the design page. It's not a good look overall especially on the homepage. Focus on only changing your design styling or revert/change the changes made to themes.tsx. (If you need help on this feel free to reach out to me or anyone else).     
- The dark brown color for the nav bar and other elements is not a good color I think it's best to keep it light grey for now like before.
- try to see if you can color coordinate or replicate the changes made to the user management table on the page (see image below). Ideally that's how the dark mode should look best. 
- Lastly the tags section on dark mode needs a small style change on how the buttons are clicked, it's hard to tell if they are being clicked because there's no highlight color being shown.  

homepage dark mode: 
![image](https://github.com/user-attachments/assets/a635eef3-46fd-45ed-af2d-6004cedc73de)

user-management-page table:
![image](https://github.com/user-attachments/assets/c834c003-3b9e-4a1b-add0-c4c6df347c49)

Overall great progress so far! Thank you

krayzieeddi commented:
Did some refactoring of CSS code for more readability.

I deleted the asterisk on the event title field and got rid of the cover photo and document fields.

I changed the tag selection section's background color to match the figma mock up.

I added the 'REQUIRED*'  text at the top of the page in red to tell users that * means required.

I got rid of the changes I implemented on the themes file, so my CSS changes will only affect the create page.

I managed to get the tags to show a contrasting color when selected on both light and dark mode but I don't like the current color as I need to inquire about the functionality more, as I don't know how to properly change the tag button colors at the moment.

Asfand00 reviewed (commented):
Appreciate the changes so far @krayzieeddi. 

Apologies, I forget to include some things I overlooked before, I'll go over them now along with new feedback: 

**Feedback (there's a lot lol):**
- I think the page looks a lot better visually. With the removed text fields, it does look a little off because there's only 2 text fields on the third column, but I suppose we can change that later (maybe move a text field from column 2 to column 3 so then its 4-3-3 for that row
---
- The tag color background is a lot better on the light side, but I feel it's too dark and it's making the tags harder to see. Try changing the tag buttons to a lighter color by adding maybe a light border around it or putting a white background inside the button (just some ideas). looks fine when its clicked though. 
---
-  I appreciate you adding the red `REQUIRED*` at the top, I think to best implement that we should follow the design and put `REQUIRED FIELDS *` instead and then for each text field add an asterisk to the fields that are required so it should look like `Event Title *`. 
---
- One of the things I forgot to say was that `Event Privacy` should be optional instead of required so no red border around that. 
---
- For each text field that's optional there should be no required message displayed in grey, so remove each one, for example Event registration has a grey message below that says its required. 
---
- The tags section is better too, some minor tweaks are needed, I think the tag section on the dark mode looks better on the light mode color wise, but when the tags are clicked it should behave the same as the light mode by turning dark blue with white text, see if you can implement that if not try to get close or ask for help from either me, @bennettsf, or someone else on the team. 
---
- As for the tag section on the light mode it mostly looks good but has an issue with the background color being too dark, we just need the tags to be visibly better somehow. 
---
- The tags on the dark mode I think are ok but need to change their color scheme when clicked on, so it looks visibly better. Try to brighten the tag text color or background or add border around it (just ideas). Also darken the background color to a dark grey maybe. I think each background color for each part should be like a dark grey color (currently blue).  
---
- finally, I managed to fix the dark brown color on the nav bar and other elements you need to change the background color on line 33 in `themes.tsx`. Since most components are using MUI paper that styling was affecting everything. Use my changes below. 
```bash
MuiPaper: {
      styleOverrides: {
        root: {
          backgroundColor: '#333',
          color: 'black'
        },
      },
    },
``` 
--- 
If I think of anymore, I will make a new post, hope all this feedback helps (sorry it was so long). Overall, this is really good progress thanks for your work @krayzieeddi.  

dark mode page: 
![image](https://github.com/user-attachments/assets/f2499fa8-45b8-4155-a572-14e67e7f482d)
light mode page: 
![image](https://github.com/user-attachments/assets/8663f9fc-d785-4f66-82a0-a9744a7a2e7f)
dark mode tag section:
![image](https://github.com/user-attachments/assets/2c137218-b2b9-40ce-a4b0-b2aaa5a3e7d7)
light mode tag section:
![image](https://github.com/user-attachments/assets/b3056eb8-5e14-4dd3-9eea-e950f86cd517)

krayzieeddi commented:
- I moved the event location field to the next box for more visual symmetry

- Changed REQUIRED text to REQUIRED FIELDS*

- The event privacy field is now optional

- All optional fields now do not have the grey required text upon missing requirements when submitting

- changed the theme file according to Ozzy's new code for the dark mode paper component

- Changed the color scheme for the tag selection buttons for light and dark mode in the hope of better contrast and visual clarity

bennettsf commented:
This is really great work! Everything on my end works well. 

The only real issue I see now is getting this to work with smaller screens/mobile. As of now, everything overlaps when it gets smaller:

![image](https://github.com/user-attachments/assets/53a65ac7-777c-4730-9b32-b32c916aaaea)

I'd be more than happy to help get this done or guide you in the right direction if you need (I had to learn how to do it when I remade the profile page). I'd like to get this PR merged before Monday if possible!

TuongPhamM reviewed (approved):
The elements are responsive and easy to look at in both dark and light mode. Beside the logic and functionality of the page, the appearance of the change meets my standards. A small change that I suggest is adding hover effect for the "create event" button in dark mode

https://github.com/user-attachments/assets/881fa27d-1521-4a00-b4f1-84e844a68d65

.

https://github.com/user-attachments/assets/a3999420-b4f4-428c-a97b-678d67d01ce4

Asfand00 reviewed (approved):
Great Work @krayzieeddi, In the interest of time I think we will try to merge these changes, its basically 95% good, just needs some minor tweaks.

**Some minor tweaks:**
- the red required text for the add tags section is a little hard to see, I think adding a white background to the text field should help for both light and dark mode.
- as @TuongPhamM suggested you could add a hover effect to "create event" on the dark mode to make it pop more, but it's a minor thing and not needed.
- last thing is adding a `(*)` to all text fields that are required which can let users know which fields need to be filled in. 

Other than that, the design looks good, and the mobile view has been fixed starting at 599 width dimension wise. 
![image](https://github.com/user-attachments/assets/737ac405-0d01-4419-b49d-4dcef2c4a0c0)

testing the page (light and dark mode):

https://github.com/user-attachments/assets/869d4a9d-e934-4138-b7cd-7741f00cd7eb

The PR seems to still have some issues regarding conflicts and a pending check not going through (if you need help feel free to reach out on this either me or @bennettsf). I will try to find time to fix this issue as well to help you out if your still stuck.

Asfand00 reviewed (changes requested):
Hey @krayzieeddi I was able to fix those conflicts, once you fix those minor tweaks I listed above in my previous comment along with this new one below, we should be good to go:

- The tags also need a hover effect on the dark mode

Thanks for your hard work.

krayzieeddi commented:
- Added hover functions to the tag, tag selectors, and create event buttons in dark mode.
- The tag section's text fields now has a white background for the light theme and a black background in dark mode
- Added *s to the required text fields

Asfand00 reviewed (changes requested):
Hey @krayzieeddi, the page itself looks and works great now, it's interesting to how you got the background box to extend when the helper text (red error message) appears. Wondering how that works, assuming it has something to do with how the theme styling is applied to all props (something like that). The hover effect color choices are something to be debated about for both light and dark mode, but I think it's fine for now in my opinion.    

I added a border radius around the tags background because it was too pointed and protruding a little bit outside the hover border effect (looked a little weird). Not sure if the other text fields need it, but at the very least the tags section needed it.

**Requested change:**
Unfortunately, there is a small issue with the font color for the homepage, maybe it was there and didn't notice before but for certain elements like inside the event card for text like date and location are appearing with black font on the dark mode. I'm wondering if there is a way to change this back to white, how it was originally. If your able to do that small change we should be good to go, otherwise if it's too difficult with how you applied the styling overall it seems, I'll need to check with @bennettsf to see if the black text is ok for certain elements, but it would be best to try and fix this if you can.

original homepage on main:
![create-event-font-change-1](https://github.com/user-attachments/assets/1b2efcd6-185a-4539-8a5a-57ef6eaf1895)

your branch homepage:
![create-event-font-change-2](https://github.com/user-attachments/assets/063ff068-a6c6-4d9b-960d-09303f5f2013)

testing your create event page:

https://github.com/user-attachments/assets/1f4d5c30-f0cd-478c-b1e5-78124732e89d

Overall, thank you for this hard work, we really appreciate it, just needs that small change if possible and we should be good for merging (If I have time I will look at it too, kind of juggling a lot of things). Get in touch with @bennettsf too to see what he thinks about the black text and your create-event page.
